private/pkg/manifest/manifest.go

```diff
@@ -214,46 +244,49 @@ func (m *Manifest) addDigest(path string, digest *Digest) error {
 // are errors from reading content, except io.EOF is swallowed.
 func (m *Manifest) AddContent(path string, content io.Reader) error {
 	m.hash.Reset()
-	if _, err := io.Copy(m.hash, content); err != nil {
-		return err
-	}
-	digest := make([]byte, shake256Length)
-	if _, err := m.hash.Read(digest); err != nil {
-		// sha3.ShakeHash never errors or short reads. Something horribly wrong
-		// happened if your computer ended up here.
+	digest, err := shake256DigestFrom(m.hash, content)
+	if err != nil {
 		return err
 	}
 	return m.addDigest(path, NewDigestFromBytes(shake256Name, digest))
 }
 
-// Paths returns one or more matching path for a given digest. The digest
-// is expected to be a lower-case hex encoded value. Returned paths are
-// unordred. paths is nil and ok is false if no paths are found.
-func (m *Manifest) Paths(digest *Digest) (paths []string, ok bool) {
-	paths, ok = m.digestToPaths[digest.String()]
-	return paths, ok
-}
-
-// Digest returns the matching digest for the given path. The path must be
-// an exact match. The returned digest is a lower-case hex encoded value.
-// digest is the empty string and ok is false if no digest is found.
-func (m *Manifest) Digest(path string) (digest *Digest, ok bool) {
-	digestValue, ok := m.pathToDigest[path]
-	digest = &digestValue
-	return digest, ok
-}
-
-// MarshalText encodes the manifest into its canonical form.
-func (m *Manifest) MarshalText() ([]byte, error) {
-	// order by path
+// Paths returns a sorted list of all paths in the manifest.
```

Comment:
I generally dislike APIs that return values with a specific order when they must do extra work to make that happen. Sorting adds significant overhead for callers who don't care about order, and especially bad (sometimes O(n^2) bad) overhead if the caller desires a different order.

I'd change this to return without a call to `sort.Strings` and have `MarshalText` sort the value.

Reply:
Done